### PR TITLE
feat: Support annotation-based column naming

### PR DESCRIPTION
**Pull request by zhixuanjia**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/utils/SQLSchemaUtils.java [hunk 1]
```diff
@@ -86,8 +91,28 @@ public static String getColumnName(@Nonnull String aspectCanonicalName) {
...
-   * Get MySQL column name from aspect class.
+   * Get Column name from aspect canonical name.
    *
+   * @param aspectCanonicalName aspect name in canonical form.
+   * @return aspect column name
+   */
+  @Nonnull
+  public static String getColumnNameFromAnnotation(@Nonnull final String aspectCanonicalName) {
+    try {
+      final RecordDataSchema schema = (RecordDataSchema) DataTemplateUtil.getSchema(Class.forName(aspectCanonicalName));
```
Comment (yangyangv2) [inline dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/utils/SQLSchemaUtils.java:102]: nit: Consider caching Class.forName()? We may get millions+ calls on this method on daily basis. 

https://stackoverflow.com/questions/18231991/class-forname-caching

File: testing/test-models/src/main/pegasus/com/linkedin/testing/AspectBar.pdl [hunk 2]
```diff
@@ -3,6 +3,7 @@ namespace com.linkedin.testing
 /**
  * For unit tests
  */
+@gma.aspect.column.name = "bar_column"
```
Comment (yangyangv2) [inline testing/test-models/src/main/pegasus/com/linkedin/testing/AspectBar.pdl:6]: is there anyway we can put some compile time validation on the 'name length'?

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/utils/SQLSchemaUtils.java (hunk 1 above)
Comment (zhixuanjia) [reply dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/utils/SQLSchemaUtils.java]: sure, added a cache.

File: testing/test-models/src/main/pegasus/com/linkedin/testing/AspectBar.pdl (hunk 2 above)
Comment (zhixuanjia) [reply testing/test-models/src/main/pegasus/com/linkedin/testing/AspectBar.pdl:6]: Validation code will be added in next PR.

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/utils/SQLSchemaUtils.java [hunk 3]
```diff
@@ -86,8 +93,37 @@ public static String getColumnName(@Nonnull String aspectCanonicalName) {
...
-   * Get MySQL column name from aspect class.
+   * Get Column name from aspect canonical name.
    *
+   * @param aspectCanonicalName aspect name in canonical form.
+   * @return aspect column name
+   */
+  @Nonnull
+  public static String getColumnNameFromAnnotation(@Nonnull final String aspectCanonicalName) {
+
+    Class aspectClass = CACHED_CLASSES.computeIfAbsent(aspectCanonicalName, className -> {
```
Comment (yangyangv2) [inline dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/utils/SQLSchemaUtils.java:104]: is CACHED_CLASSES.computeIfAbsent thread safe?

might be worth of the efforts to create a global className class. You may improve it in the next PR.

**Review by yangyangv2 — APPROVED** [review]
LGTM

**Pull request merged**